SeriesCruncher/app/src/main/java/com/balazs_csernai/seriescruncher/utils/color/ColorProviderImpl.java

```diff
@@ -36,52 +38,101 @@ public void generateColorPalette(Bitmap bitmap) {
         if (bitmap != null) {
             List<Swatch> swatches = Palette.from(bitmap).generate().getSwatches();
             if (swatches.isEmpty()) {
-                swatch = null;
+                primarySwatch = null;
+                secondarySwatch = null;
             } else {
-                swatch = findMostCommonSwatch(swatches);
+                primarySwatch = findPrimaryCommonSwatch(swatches);
+                secondarySwatch = findSecondarySwatch(swatches);
             }
         }
     }
 
-    private Swatch findMostCommonSwatch(@NonNull List<Swatch> swatches) {
+    private Swatch findPrimaryCommonSwatch(@NonNull List<Swatch> swatches) {
         Iterator<Swatch> iterator = swatches.iterator();
-        Swatch swatch, mostCommonSwatch = iterator.next();
+        Swatch swatch, primarySwatch = iterator.next();
+
+        while (iterator.hasNext()) {
+            swatch = iterator.next();
+            if (swatch.getPopulation() > primarySwatch.getPopulation()) {
+                primarySwatch = swatch;
+            }
+        }
+        return primarySwatch;
+    }
+
+    private Swatch findSecondarySwatch(@NonNull List<Swatch> swatches) {
+        Iterator<Swatch> iterator = swatches.iterator();
+        Swatch swatch, secondarySwatch = iterator.next();
+
+        // Make sure that the primary swatch is not selected as secondary swatch
+        if (secondarySwatch.getPopulation() == primarySwatch.getPopulation()) {
```

Comment:
Just an idea: if you would remove the primarySwatch from the list of swatches passed to the findSecondarySwatch() method, then you would not need this check, and the isPopulationHighestAfterPrimary() check in the while loop below.